app.js

```diff
@@ -2,17 +2,64 @@ require('dotenv').config();
 
 const express = require('express');
 const hbs = require('hbs');
+const SpotifyWebApi = require('spotify-web-api-node');
+const spotifyApi = new SpotifyWebApi({
+    clientId: process.env.CLIENT_ID,
+    clientSecret: process.env.CLIENT_SECRET
+  });
+
+
 
-// require spotify-web-api-node package here:
 
 const app = express();
 
 app.set('view engine', 'hbs');
 app.set('views', __dirname + '/views');
 app.use(express.static(__dirname + '/public'));
 
-// setting the spotify-api goes here:
+spotifyApi
+  .clientCredentialsGrant()
+  .then(data => spotifyApi.setAccessToken(data.body['access_token']))
+  .catch(error => console.log('Something went wrong when retrieving an access token', error));
+
+
+app.get('/', (req,res)=>{
+    res.render('search')
+})
+
+app.get('/artist-search',(req,res)=>{
+    console.log('artist getting searched')
+    spotifyApi
+  .searchArtists(req.query.q)
+  .then(data => {
+    console.log(data.body);
+    const searchedMusic = data.body;
+    res.render('music', {searchedMusic});
+  })
+  .catch(err => console.log('error while searching artists occurred: ', err));
+})
+
+app.get('/albums/:artist', (req, res, next)=>{
+    spotifyApi
+    .getArtistAlbums(req.params.artist.substr(15))
```

Comment:
If you passed the id of the artist from the hbs file, you can use it here directly without `substr()`.